### ISSUE TITLE
Full Eora

### ISSUE THREAD

**Issue by sanojb99** [konstantinstadler/pymrio#45]
Hello and thank you for providing _Pymrio_ for free and open source!
I am wondering why the _full_ version of Eora is not supported by Pymrio? I am working with the full Eora database and I am happy to write the code for the calculations myself. It helps me to understand what I am actually doing. Still, I noticed that the full Eora is not supported by Pymrio and I am simply curios what the reason is.
Thank you and best regards
Jonas

**Comment by konstantinstadler** [comment]
Hi Jonas,

Short answer: I never worked with Eora (being actively involved in the Exiobase development) and it was easiest to handle all the consistent MRIOs (in regard to sector classifications) first.

Anyhow, mathematically I do not see any problems using pymrio for Eora. There are some functions which need to be adopted (e.g. get_sectors which assumes consistent classification) and one would need to carefully go through the diagonlization of stressors procedures and the aggregations functions (I believe the later would be the most tedious one as it might also imply some changes in country_converter). Principally, the way forward would be to construct a (small) mock Eora (perhaps just the test_mrio with some varying sector classifications, but one would need to go through Eora to capture all peculiarities of it in the Eora mock) for development and, importantly, testing.

More generally, I never believed pymrio has a use case for Eora, given the virtual lab environment developed by the same team. But I am happy to stand corrected. Currently I am on paternity leave and will be back in autumn.  I have some funding to extend pymrio so if I see the need I might start including the full Eora in the next half year - year or so. But I am of course more than happy to get some help. There is a contributing.rst which gives you some first starting points.

Happy to discuss further, but expect some delays in my response (back at work in September)

best
kst

**Comment by sanojb99** [comment]
Hi Konstantin,
thank you very much four your elaborate response which sounds quite reasonable!

I wasn't aware that the Eora team provides a _virtual lab environment_. Searching on Google I came across [https://ielab.info/resources](https://ielab.info/resources) but it's empty. I also cannot find any information on the Eora website. Maybe the project is discontinued?

I just took a brief look at `pymrio/tools/iomath.py`. It's very well written. Clear and easy to understand! Nice for beginners like me. I'll check later this year whether I find time to contribute. Also, I have a colleague working on Structural Decomposition Analysis which I saw is on your to-do list. I'll point him towards Pymrio. Maybe he is interested in contributing.

Regarding two more points on your to-do list:

* [floWeaver](https://github.com/ricklupton/floweaver) is an excellent package for flow maps
* Chloropleth maps in [GeoPandas](geopandas.org/) are as easy as `GeoDataFrame.plot(column='footprint', legend=True)`
